error TS2303: Circular definition of import alias 'default'.

Hi, I'm switching to ES6 modules. The type definition below worked fine in 1.5.0-alpha, but gives an error in 1.5.0-beta

```
declare var io: SocketIOClientStatic;

declare module 'socket.io-client' {
    export default io;
}
```

It seems that error TS2303: Circular definition of import alias 'default' is always given when the export default is a variable. At least the error message is misleading. The earlier form 'export = io' is not allowed in ES6 output mode.

repo:

```ts
declare var io: any;

declare module 'module' {
    export default io;
}
```
error TS2303: Circular definition of import alias 'default'.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -350,9 +350,9 @@ module ts {
                         }
                         else if (location.kind === SyntaxKind.SourceFile ||
                             (location.kind === SyntaxKind.ModuleDeclaration && (<ModuleDeclaration>location).name.kind === SyntaxKind.StringLiteral)) {
-                            result = getSymbol(getSymbolOfNode(location).exports, "default", meaning & SymbolFlags.ModuleMember);
+                            result = getSymbolOfNode(location).exports["default"];
                             let localSymbol = getLocalSymbolForExportDefault(result);
-                            if (result && (result.flags & meaning) && localSymbol && localSymbol.name === name) {
+                            if (result && localSymbol && (result.flags & meaning) && localSymbol.name === name) {
                                 break loop;
                             }
                             result = undefined;

--- a/src/compiler/utilities.ts
+++ b/src/compiler/utilities.ts
@@ -1695,7 +1695,7 @@ module ts {
     }
 
     export function getLocalSymbolForExportDefault(symbol: Symbol) {
-            return symbol && symbol.valueDeclaration && (symbol.valueDeclaration.flags & NodeFlags.Default) ? symbol.valueDeclaration.localSymbol : undefined;
+        return symbol && symbol.valueDeclaration && (symbol.valueDeclaration.flags & NodeFlags.Default) ? symbol.valueDeclaration.localSymbol : undefined;
     }
 
     /**

--- a/tests/baselines/reference/exportDefaultVariable.js
+++ b/tests/baselines/reference/exportDefaultVariable.js
@@ -1,0 +1,12 @@
+//// [exportDefaultVariable.ts]
+// Regression test for #3018
+
+declare var io: any;
+
+declare module 'module' {
+    export default io;
+}
+
+
+//// [exportDefaultVariable.js]
+// Regression test for #3018

--- a/tests/baselines/reference/exportDefaultVariable.symbols
+++ b/tests/baselines/reference/exportDefaultVariable.symbols
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/exportDefaultVariable.ts ===
+// Regression test for #3018
+
+declare var io: any;
+>io : Symbol(io, Decl(exportDefaultVariable.ts, 2, 11))
+
+declare module 'module' {
+    export default io;
+>io : Symbol(default, Decl(exportDefaultVariable.ts, 2, 11))
+}
+

--- a/tests/baselines/reference/exportDefaultVariable.types
+++ b/tests/baselines/reference/exportDefaultVariable.types
@@ -1,0 +1,11 @@
+=== tests/cases/compiler/exportDefaultVariable.ts ===
+// Regression test for #3018
+
+declare var io: any;
+>io : any
+
+declare module 'module' {
+    export default io;
+>io : any
+}
+

--- a/tests/cases/compiler/exportDefaultVariable.ts
+++ b/tests/cases/compiler/exportDefaultVariable.ts
@@ -1,0 +1,7 @@
+// Regression test for #3018
+
+declare var io: any;
+
+declare module 'module' {
+    export default io;
+}

